### PR TITLE
Add fp8 mqa logits kernels to support sparse MLA based models

### PR DESCRIPTION
**Pull request by YangQun1**
# Essential Elements of an Effective PR Description Checklist

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.

PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS ABOVE HAVE BEEN CONSIDERED.

## Purpose

Add fp8 mqa logits (both paged and non-paged variants) to support sparse MLA based models

## Test Plan

pytest tests/mqa_logits/test_mqa_logits.py -sv
python benchmark/benchmark_mqa_logits.py
python benchmark/benchmark_mqa_logits.py --mode paged

## Test Result

Passed

## (Optional) Documentation Update

**BEFORE SUBMITTING, PLEASE READ <https://docs.vllm.ai/en/latest/contributing>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: csrc/xpu/mqa_logits/xe_2/mqa_logits_xe2.cpp [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+    float output = 0;
+    const int64_t kv_index = kv_block_start + local_id;
+    const float scale = kv_index < seq_len_kv ? scales_tensor(kv_index) : 0.0f;
+    float* weights_smem = reinterpret_cast<float*>(smem_buf);
+
+    for (int64_t head_tile = 0; head_tile < head_tile_count; head_tile++) {
+      int k_tile_prefetch = 0;
+      clear(tCrC);
+
+      barrier_arrive(barrier_scope);
```
Comment (wuxun-zhang) [inline csrc/xpu/mqa_logits/xe_2/mqa_logits_xe2.cpp:200]: why we need barrier here?

File: tests/mqa_logits/test_mqa_logits.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+                block_rk * block_size:(block_rk + 1) * block_size,
+            ] = torch.where(
+                k_offsets[None, :] <= q_offsets[:, None],
+                s,
+                float("-inf"),
+            )
+    return logits
+
+
+@pytest.mark.parametrize("seq_len_qkv", [(512, 1024), (24, 24)])
```
Comment (wuxun-zhang) [inline tests/mqa_logits/test_mqa_logits.py:175]: Add `MINI_PYTEST_PARAMS` test for simulator testing?

**Comment by jikunshang** [comment]
is it necessary to add another lib for this kernel?

File: csrc/xpu/mqa_logits/xe_2/mqa_logits_xe2.cpp (hunk 1 above)
Comment (YangQun1) [reply csrc/xpu/mqa_logits/xe_2/mqa_logits_xe2.cpp:200]: to make sure smem weights are fully comsumed before we begin next iteration

**Comment by YangQun1** [comment]
> is it necessary to add another lib for this kernel?

Hi @jikunshang I just followed other kernels like grouped_gemm/gdn_atn/... to add a separated lib. So, what is the criteria to decide whether add a lib or not?

**Comment by jikunshang** [comment]
oh, this is also a sycl-tla based kernel... 
we have flash attention and moe gemm at very first, each are a standalone lib and will be common used. then we add gdn attention for specific model(qwen3-next/qwen3.5), which also use sycl-tla. I am considering should we consolidate such kernels into one lib which may avoid some potential issue.
cc @mayuyuace @xinyu-intel

**Comment by mayuyuace** [comment]
@jikunshang 
I think we can push all sycl-tla kernels into one lib.

**Comment by wuxun-zhang** [comment]
Just a thought, by merging into a single lib, will it lead to more linking time? For example, if only touch FA related files, besides those FA related object files, the final .so file will need to link other object files again.

**Comment by jikunshang** [comment]
I think we should keep fa/moe as standalone one. other sycl-tla could put in one lib. this should not increase build/jit time. we can make it parallel build.

**Comment by xinyu-intel** [comment]
> I think we should keep fa/moe as standalone one. other sycl-tla could put in one lib. this should not increase build/jit time. we can make it parallel build.

Shall we go ahead with this PR or merge the lib merge?

**Comment by wuxun-zhang** [comment]
Suggest to move forward this PR firstly and then open another PR for lib merge if necessary.

Regards lib merge, I think mqa_logits kernels should be along with MLA/SparseMLA kernels.

**Review by xinyu-intel — APPROVED** [review]

**Review by wuxun-zhang — APPROVED** [review]
LGTM

**Review by jikunshang — APPROVED** [review]

File: csrc/xpu/mqa_logits/xe_2/mqa_logits_xe2.cpp [hunk 3]
```diff
@@ -0,0 +1,690 @@
...
+
+  const auto* q_ptr = reinterpret_cast<const float_e4m3_t*>(q.data_ptr());
+  const auto* kv_ptr = reinterpret_cast<const float_e4m3_t*>(kv.data_ptr());
+  const float* scales_ptr = kv_scales.data_ptr<float>();
+  const float* weights_ptr = weights.data_ptr<float>();
+  const int32_t* ks_ptr = cu_seqlen_ks.data_ptr<int32_t>();
+  const int32_t* ke_ptr = cu_seqlen_ke.data_ptr<int32_t>();
+  float* out_ptr = logits.data_ptr<float>();
+
+  auto queue = c10::xpu::getCurrentXPUStream().queue();
```
Comment (jikunshang) [inline csrc/xpu/mqa_logits/xe_2/mqa_logits_xe2.cpp:530]: should be a lvalue reference, I would suggest use `vllmGetQueue`
